pallets/allocations/src/lib.rs

```diff
@@ -333,42 +361,57 @@ impl<T: Config> Pallet<T> {
 		Ok(())
 	}
 
-	/// Calculate the session quota and update the corresponding storage only at the beginning of a
-	/// fiscal period.
-	/// Return the weight of itself.
-	fn checked_calc_session_quota(n: T::BlockNumber, forced: bool) -> Weight {
-		if let Some(session_quota) =
-			T::MintCurve::get().checked_calc_next_session_quota(n, T::Currency::total_issuance(), forced)
-		{
-			<NextSessionQuota<T>>::put(session_quota);
-			Self::deposit_event(Event::SessionQuotaCalculated(session_quota));
-			T::WeightInfo::calc_quota()
-		} else {
-			T::DbWeight::get().reads(1 as Weight) // Storage: Balances TotalIssuance (r:1 w:0)
+	fn checked_update_session_quota() {
```

Comment:
Slightly longer functions would improve readability, it is not obvious which are pure functions and where there are side effects.

Reply:
All the pallet functions have access to the pallet storage and regardless of their look they can not be considered pure functions, it's substrate design. Only the MintCurve functions don't have access to the pallet storage and there is only one pure function called MintCurve::next_schedule.